Remotion/Web/Development.WebTesting/CoypuElementScopeExtensions.cs

```diff
@@ -40,27 +40,56 @@ public static class CoypuElementScopeExtensions
     /// where the <c>parentScope.Find*()</c> call could not be found.
     /// </summary>
     /// <param name="scope">The <see cref="ElementScope"/> which is asserted to exist.</param>
+    /// <exception cref="WebTestException">If the control cannot be found.</exception>
+    /// <exception cref="WebTestException">If multiple matching controls are found.</exception>
     public static void EnsureExistence ([NotNull] this ElementScope scope)
     {
       ArgumentUtility.CheckNotNull ("scope", scope);
 
-      scope.Now();
+      try
+      {
+        scope.Now();
+      }
+      catch (MissingHtmlException exception)
+      {
+        throw new WebTestException ($"The element cannot be found: {exception.Message}");
+      }
+      catch (AmbiguousException exception)
+      {
+        throw new WebTestException ($"Multiple elements were found: {exception.Message}");
+      }
     }
 
     /// <summary>
     /// Ensures that the given <see cref="Scope"/> exists, much like <see cref="EnsureExistence"/>. However, it ensures that Coypu uses the
     /// <see cref="Match.Single"/> matching strategy.
     /// </summary>
     /// <param name="scope">The <see cref="ElementScope"/> which is asserted to match only a single DOM element.</param>
+    /// <exception cref="WebTestException">If the control cannot be found.</exception>
+    /// <exception cref="WebTestException">If multiple matching controls are found.</exception>
     public static void EnsureSingle ([NotNull] this ElementScope scope)
     {
       ArgumentUtility.CheckNotNull ("scope", scope);
 
       var matchBackup = scope.ElementFinder.Options.Match;
 
-      scope.ElementFinder.Options.Match = Match.Single;
-      scope.Now();
-      scope.ElementFinder.Options.Match = matchBackup;

```

Comment:
looks like this was buggy before?

Reply:
buggy in what way?

Reply:
there was no setting the match back to the backup in the event of an exception before this change.